### ISSUE TITLE
Throttling problems

### ISSUE THREAD

**Issue by elibarzilay** [Azure/azure-cli#4229]
We have a build that relies heavily on using azure storage via the CLI.
A major thing that bugged me is that from time to time I get errors due
to throttling.  Recently this turned from an occasional annoyance to a
major issue, since we now have builds that fail because of throttle
related problems.

Solving these problems in a robust way as far as I can currently see is
difficult enough[*] that it's practically impossible.  There should
really be a flag for "just make it work, wait whatever time it takes for
it".

I think that such a mode should be the default, but just a flag would be
a huge help.

[*] The way I could think about resolving it now is to run the command
and capture its stdout and stderr separately; inspect the stderr for
things that *look* like throttle problems (ie, check for a magic word in
the stderr); and if there is such a problem, guess some time to sleep
and re-try (sometimes there's information in the stderr, but parsing
that would be very brittle); if not then spit out the saved stdout.

**Comment by elibarzilay** [comment]
See also Issue #4347

When I was trying to trace how things work, it turns out that the CLI is
actually doing the right thing (via python http library): it gets a
reply from the management API that asks to re-try after 17 seconds, and
then it actually *does* that 17 second sleep, and retries.  In some
cases this works, and in other cases it just does this three times and
then it fails.

So this looks like an issue with the API not returning a proper header,
since the CLI is definitely doing what it should.  There's of course the
chance that someone else was hammering at the subscription at the same
time, but I have been tracking this problem for several days, and in all
that time we got consistent throttling, and the requested wait time was
*always* 17 seconds.

Bottom line: this issue should probably be closed, but you might want to
raise an issue with the API about that problem.  (Not the low threshold,
but the fact that it lies about the 17 second sleep...)
